### PR TITLE
Immediately clean again if work was done last time.

### PR DESCRIPTION
**Pull request by AlCutter**
image
Issue #664

### CODE REVIEW

File: cpp/server/ct-server.cc [hunk 1]
```diff
@@ -237,11 +237,22 @@ void CleanUpEntries(ConsistentStore<LoggedCertificate>* store,
...
-          sequencer_delete_latency_ms.ScopedLatency());
-      util::Status status(store->CleanupOldEntries());
-      if (!status.ok()) {
-        LOG(WARNING) << "Problem cleaning up old entries: " << status;
+      // Keep cleaning up until there's no more work to do.
+      // This should help to keep the etcd contents size down during heavy
+      // load.
+      while (true) {
+        const ScopedLatency sequencer_delete_latency(
+            sequencer_delete_latency_ms.ScopedLatency());
```
Comment (pphaneuf) [inline cpp/server/ct-server.cc]: This covering a possibly unbounded time seems a bit iffy to me...

**Comment by pphaneuf** [comment]
Just the one comment.

File: cpp/server/ct-server.cc (hunk 1 above)
Comment (AlCutter) [inline cpp/server/ct-server.cc]: Yeah, fair point - there's still the op latency inside the method itself, which is probably more useful.

**Comment by pphaneuf** [comment]
LGTM

**Pull request merged**